test/generate-xml.sh

```diff
@@ -105,3 +113,20 @@ for hexfile in "$directory"/*.hex;  do
     generate_xml "$directory" "$hexfile" "normalized"
 done
 
+# Check the size of the file $FAILING_TESTS. Make sure to indicate failure.
+if [ -s $FAILING_TESTS ]; then
+    echo "** There were errors in the following file(s):"
+    ls -l $FAILING_TESTS
+    cat $FAILING_TESTS
+    exit 1
+else
+    rm $FAILING_TESTS
+fi
+
+# Check that there was exactly the correct number of files that failed to parse
+EXPECTED_PARSE_ERRORS=4
```

Comment:
The fact that we expect parse errors is bad. I suggest to move the relevant hex files into unsupported-frames.

Reply:
There can be a point in doing negative tests.
I see that we have two more directories, unsupported-frames and error-frames.

Should we do execute the code for these directories as well?

Since the files are named: `invalid_length.hex` and `invalid_length2.hex`, maybe `error-frames/` is more appropriate?

Reply:
Yes, negative tests are a good point. The only one we currently have are invalid_length*.hex
If you want to integrate this in the pull request make a two different directories like test-frames-good and test-frames-bad. But if you go this way, this logic should good out of generate-xml.sh

error-frames are valid errors messages send by the slave. The mentioned hex files doesn't fit in there. unsupported-frames are for currently unsupported frames so executing code on these directories doesn't make sense to me.

Reply:
I added a simple if statement for the names of the directories that should contain errors.
It seem simplest to keep this information in this script, otherwise we would have some other script know that error-frames should have 30 parse errors, and pass that information to generate-xml.sh.

See 62b5829 and 50e3141.

Reply:
The fact that we code the expectation of x parse errors is the whole problem. I don't want this a script to generate XML files. Please keep it simple and easy to maintain. Also i don't want that we run generate-xml on the unsupported frames as a test. Except of that i'm happy to pull.